pkg/planner/core/find_best_task.go

```diff
@@ -1391,6 +1399,19 @@ func findBestTask4LogicalDataSource(lp base.LogicalPlan, prop *property.Physical
 		planCounter.Dec(1)
 		return nil, 1, nil
 	}
+	// if prop is require an index join's probe side, check the inner pattern admission here.
```

Comment:
Datasource's findBestTask portal should feel the index join prop and build index range info based on chosen path, and return the inner task back attached with indexJoinInfo.